### ISSUE TITLE
deno lsp panic 

### ISSUE THREAD

**Issue by sigmaSd** [denoland/deno#18712]
```
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: linux x86_64
Version: 1.32.4
Args: ["/home/mrcool/.deno/bin/deno", "lsp"]

thread '<unnamed>' panicked at 'assertion failed: `(left == right)`
  left: `Some(TestDescription(TestDescription { id: 0, name: "list", origin: "file:///home/mrcool/dev/deno/chef/src/lib.ts", location: TestLocation { file_name: "file:///home/mrcool/dev/deno/chef/src/lib.ts", line_number: 282, column_number: 6 } }))`,
 right: `Some(TestStepDescription(TestStepDescription { id: 1, name: "function prints all key-value pairs in database", origin: "file:///home/mrcool/dev/deno/chef/src/lib.ts", location: TestLocation { file_name: "file:///home/mrcool/dev/deno/chef/src/lib.ts", line_number: 285, column_number: 11 }, level: 1, parent_id: 0, root_id: 0, root_name: "list" }))`', cli/lsp/testing/execution.rs:817:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I think this happens when I run the tests from vscode interface 
```js
import { assertEquals } from 'https://deno.land/std/testing/asserts.ts';
import {
  assertSpyCall,
  spy,
} from "https://deno.land/std@0.159.0/testing/mock.ts";

Deno.test('list', async (t) => {
  const spyLog = spy(console, 'log');

  await t.step("function prints all key-value pairs in database", () => {
  const mockDbData = { program1: 'Hello', program2: 'world' };
  const mockDb = {
    readDb: () => mockDbData,
  };
  ChefInternal.prototype.list.call(mockDb);
  assertSpyCall(spyLog, 0, {
      args:["%cprogram1 %cHello","color: #ffff00","color: #00ff00"]
  });
});

await t.step("fn handle empty database", () => {
  const mockDb = {
    readDb: () => {return{}},
  };
  ChefInternal.prototype.list.call(mockDb);
  assertSpyCall(spyLog, 0, {
    args:["No recipes yet"]
  });
})
})
```
I can share the code if its needed

**Comment by jtiai** [comment]
I hit the very same issue.

I changed my code so that I removed steps and made them as normal `Deno.test()` calls and no panic was raised.

**Comment by sigmaSd** [comment]
I forgot how to reproduce this, maybe you can create  a minimal example @jtiai

**Comment by jtiai** [comment]
```ts
import { assertEquals } from "https://deno.land/std@0.189.0/testing/asserts.ts";
Deno.test("Test steps", async (t) => {
   await t.step("step 1", () => { assertEquals(1, 1); }:
   await t.step("step 2", () => { assertEquals(1, 2); }:
}
```

**Comment by nayeemrmn** [comment]
Fixed by #20289.
